### PR TITLE
connected forgot password to backend and created reset password form …

### PR DESCRIPTION
**Pull request by DwinaTech**
connected forgot password form with forgot API endpoint then created reset form and connected to reset API endpoints and used notification component to display the messages.

### CODE REVIEW

File: frontend/src/Components/Users/forgot-password/index.js [hunk 1]
```diff
@@ -1,20 +1,62 @@
...
+      message,
+      level: 'error',
+    });
   };
 
   handleSubmit = e => {
     e.preventDefault();
-    this.setState({email: ''});
+    const { email } = this.state;
+    const sideHost = process.env.SIDE_HOST
```
Comment (kkarimi) [inline frontend/src/Components/Users/forgot-password/index.js]: what is side? is it meant to be site?

File: frontend/src/actions/postData.js [hunk 2]
```diff
@@ -13,3 +13,27 @@ export function editOrganisation(data) {
     axios.patch(`${api}/service/organisation/edit`, data).then(res => res);
   return saveOrganisation
 }
+
+export function requestRestPassword(data) {
+  const requestFun = () =>
+    axios.post(`${api}/forgot`, data)
```
Comment (kkarimi) [inline frontend/src/actions/postData.js]: No try/catch, no await.. can we be consistent?

File: frontend/src/Components/Users/forgot-password/index.js (hunk 1 above)
Comment (DwinaTech) [reply frontend/src/Components/Users/forgot-password/index.js]: I meant the the application domain/host name

**Comment by kkarimi** [comment]
@DwinaTech still no try/catch for promises

**Comment by kkarimi** [comment]
@DwinaTech I do not understand how this is possible? can you push the code you tried so we can see the problem?

**Comment by kkarimi** [comment]
@DwinaTech when a code crashes you need to investigate what error you got, and why it was not caught by try/catch. Researching is very important, what did you find? did you not find anyone else having the same problem as you? because I can tell you from 1 quick google search it is possible to know why you are having this issue.

**Pull request merged**